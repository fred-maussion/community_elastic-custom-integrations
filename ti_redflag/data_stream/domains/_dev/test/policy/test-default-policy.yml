input: cel
data_stream:
  vars:
    dataset: ti_redflag.domains
    interval: 12h
    url: "https://dl.red.flag.domains/red.flag.domains.txt"
    tags:
      - red-flag-domains
      - forwarded

