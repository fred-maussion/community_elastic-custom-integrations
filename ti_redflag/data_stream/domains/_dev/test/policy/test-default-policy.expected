inputs:
    - data_stream:
        namespace: ep
      meta:
        package:
            name: ti_redflag
            policy_template: redflag_domains
            release: beta
      name: test-default-policy-ti_redflag
      streams:
        - data_stream:
            dataset: ti_redflag.domains
          interval: 12h
          program: |-
            state.with(
              request(
                "GET",
                state.url
              ).do_request().as(resp,
                (resp.StatusCode == 200) ? {
                  "events": string(resp.Body)
                    .split('\n')
                    .filter(line, line != '' && !line.startsWith('#'))
                    .map(domain, {"message": domain})
                } : {
                  "events": [{
                    "message": "ERROR: Failed to download Red Flag Domains feed. Status code: " + string(resp.StatusCode)
                  }]
                }
              )
            )
          resource.timeout: 30s
          resource.url: https://dl.red.flag.domains/red.flag.domains.txt
          state:
            url: https://dl.red.flag.domains/red.flag.domains.txt
          tags:
            - red-flag-domains
            - forwarded
      type: cel
      use_output: default
output_permissions:
    default:
        _elastic_agent_checks:
            cluster:
                - monitor
        _elastic_agent_monitoring:
            indices: []
        uuid-for-permissions-on-related-indices:
            indices:
                - names:
                    - logs-ti_redflag.domains-ep
                  privileges:
                    - auto_configure
                    - create_doc
secret_references: []
